### PR TITLE
fixes using webpack harmony modules from babel

### PR DESCRIPTION
**Pull request by sokra**
**What kind of change does this PR introduce?**

**Did you add tests for your changes?**

**If relevant, link to documentation update:**

**Summary**
see #3168

**Does this PR introduce a breaking change?**

**Other information**
fixes #3268

### CODE REVIEW

**Comment by zhaoshengjun** [comment]
This is so different than my initial change! Good to learn your way to handle this :smile: 

Thanks!

File: test/cases/parsing/harmony-commonjs/index.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+it("should pass when required by CommonJS module", function () {
+	var test1 = require('./a').default;
+	test1().should.be.eql("OK");
+});
+
+it("should pass when use babeljs transpiler", function() {
+	//the following are generated code by use babeljs.
+	// use it this way will save trouble to setup babel-loader
+	// the babeljs transpiled code depends on the __esMoudule to be set
```
Comment (Kovensky) [inline test/cases/parsing/harmony-commonjs/index.js]: Maybe

> The following is code as generated by babel's commonjs transform.
> This is how other code importing the bundle may try to use the exports,
> they depend on __esModule being set to load it correctly.

File: lib/dependencies/HarmonyCompatiblilityDependency.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+HarmonyCompatiblilityDependency.prototype = Object.create(NullDependency.prototype);
+HarmonyCompatiblilityDependency.prototype.constructor = HarmonyCompatiblilityDependency;
+HarmonyCompatiblilityDependency.prototype.type = "harmony export header";
+
+HarmonyCompatiblilityDependency.Template = function HarmonyExportDependencyTemplate() {};
+
+HarmonyCompatiblilityDependency.Template.prototype.apply = function(dep, source) {
+	var usedExports = dep.originModule.usedExports;
+	if(usedExports && !Array.isArray(usedExports)) {
```
Comment (Kovensky) [inline lib/dependencies/HarmonyCompatiblilityDependency.js]: I suppose this is only the case in the entry point?

Comment (sokra) [inline lib/dependencies/HarmonyCompatiblilityDependency.js]: no every time a module is not exclusivly used by webpack harmony modules.

**Comment by jquense** [comment]
was just about to open a PR for this glad to see it's already fixed! :)

**Pull request merged**